File: app/interfaces/api/boundedContext/json/ProjectResponse.scala
package interfaces.api.boundedContext.json

import dev.tchiba.sdmt.core.models.boundedContext.BoundedContext
import interfaces.json.PlayJsonResponse
import play.api.libs.json.{JsValue, Json, OFormat}

import java.util.UUID

final class ProjectResponse private (project: BoundedContext) extends PlayJsonResponse {

  import ProjectResponse._

  private val response = Response(
    projectId = project.id.value,
    projectAlias = project.alias.value,
    projectName = project.name.value,
    projectOverview = project.overview.value
  )

  override def json: JsValue = toJson(response)
}

object ProjectResponse {
  def apply(project: BoundedContext) = new ProjectResponse(project)

  private case class Response(
      projectId: UUID,
      projectAlias: String,
      projectName: String,
      projectOverview: String
  )
  implicit private val jsonFormat: OFormat[Response] = Json.format[Response]
}

File: front/angular-front/src/app/models/boundedContext/http/ProjectResponse.ts
import {BoundedContext} from "../bounded-context";
import {BoundedContextId} from "../id/bounded-context-id";
import {BoundedContextAlias} from "../alias/bounded-context-alias";
import {BoundedContextName} from "../name/bounded-context-name";
import {BoundedContextOverview} from "../overview/bounded-context-overview";


export type ProjectResponse = {
  readonly projectId: string
  readonly projectAlias: string
  readonly projectName: string
  readonly projectOverview: string
}

export const ProjectResponse = {
  convert(res: ProjectResponse): BoundedContext {
    return new BoundedContext(
      new BoundedContextId(res.projectId),
      new BoundedContextAlias(res.projectAlias),
      new BoundedContextName(res.projectName),
      new BoundedContextOverview(res.projectOverview)
    )
  }
}

